Packs/BitwardenPasswordManager/Integrations/BitwardenPasswordManager/BitwardenPasswordManager.py

```diff
@@ -0,0 +1,236 @@
+        new_last_fetch_date = max([dt for dt in (arg_to_datetime(event.get('date'), DATE_FORMAT)
+                                                 for event in unique_events) if dt is not None]).strftime(
+            DATE_FORMAT) if unique_events else get_current_time()
+        new_last_run = {'last_fetch': new_last_fetch_date, 'hashed_latest_events': hash_latest_events}
+        demisto.debug(f'Bitwarden - Fetched {len(unique_events)} events')
+
+    for event in unique_events:
+        event['_time'] = event.get('date')
+
+    return unique_events, new_last_run
+
+
+def get_events(client: Client, max_fetch: int, start_date_str: str, last_run: Dict[str, Any]) -> tuple[List[Dict[str, Any]], str]:
+    continuation_token = last_run.get('continuationToken', '')
+    events: List[dict] = []
+    has_next = True
+    while has_next:
+        has_next = False
+        if len(events) >= max_fetch:
+            break
+        start_date = last_run.get('last_fetch', '') if last_run.get('last_fetch', '') else start_date_str
+        response = client.get_events(start_date=start_date, continuation_token=continuation_token)
+        if continuation_token := response.get('continuationToken'):
+            has_next = True
+        events.extend(response.get('data'))
+    return events, continuation_token
+
+
+def get_unique_events(events: List[Dict[str, Any]], last_run: Dict[str, Any]) -> List[Dict[str, Any]]:
+    if hashed_latest_events := last_run.get('hashed_latest_events'):
+        recent_events = filter_events(events=events, latest=False)
+        hashed_recent_events = hash_events(recent_events)
+        should_be_removed_events = []
+        for hashed_recent_event in hashed_recent_events:
+            if list(hashed_recent_event.keys())[0] in hashed_latest_events:
+                should_be_removed_events.append(list(hashed_recent_event.values())[0])
+
+        events_to_remove_set = {tuple(event.items()) for event in should_be_removed_events}
+        unique_events = [event for event in events if tuple(event.items()) not in events_to_remove_set]
+        return unique_events
+    return events
+
+
+def filter_events(events: List[Dict[str, Any]], latest: bool) -> List[Dict[str, Any]]:
+    sorted_events = sorted(events, key=lambda x: x['date'])
+    date = sorted_events[0]['date'] if latest else sorted_events[-1]['date']
+    filtered_events = [event for event in sorted_events if event['date'] == date]
+
+    return filtered_events
+
+
+def hash_events(events: List[Dict[str, Any]]) -> List[Dict[str, Dict[str, Any]]]:
+    hashed_events = []
+    for event in events:
+        event = {key: "null" if value is None else value for key, value in event.items()}
+        event_str = json.dumps(event, sort_keys=True)
+        event_hash_object = hashlib.sha256(event_str.encode()).hexdigest()
+        hashed_events.append({event_hash_object: event})
```

Comment:
a single dict will work better for you than a list of dict with a single item each